### ISSUE TITLE
Rvision installation: non-zero exit status

### ISSUE THREAD

**Issue by mirickmi** [swarm-lab/Rvision#58]
**Describe the bug**
The installation of Rvision 0.6.2 is not working. The following error message  (translated to english) shows up:

Error: [...] cannot load shared object 'C:/Users/XXX/Documents/R/win-library/4.1/Rvision/libs/i386/Rvision.dll':
  LoadLibrary failure:  The module was not found. 

Error: [...] cannot load shared object 'C:/Users/XXX/Documents/R/win-library/4.1/Rvision/libs/x64/Rvision.dll':
  LoadLibrary failure:  The module was not found.

Error: loading failed
Execution stopped
ERROR: loading failed for 'i386', 'x64'
* removing 'C:/Users/XXX/Documents/R/win-library/4.1/Rvision'
Warning message:
In i.p(...) :
  installation of package ‘C:/Users/XXX/AppData/Local/Temp/Rtmpst2r9Q/file368c6d5e49f9/Rvision_0.6.2.tar.gz’ had non-zero exit status

**To Reproduce**
As per installation guide I installed the most recent versions of the following:
CMake 3.23.2
RTools 4.0.3.0
ROpenCVLite 4.60.2
ROpenCV 4.6.0

Interestingly, before the installation stopped and the Rvision folder was removed once again I checked the following path:
C:/Users/XXX/Documents/R/win-library/4.1/Rvision/libs/

And I did find the Rvision.dll in both the 'x64' and the 'i386' folders. 

**Expected behavior**
-

**Screenshots**
-

**Computer (please complete the following information):**
 - Operating system: Windows 10 
 - Version 21H2

**R (please complete the following information):**
 - R version: 4.1.1
 - Rvision version: 0.6.2
 - ROpenCVLite version: 4.60.2

**Additional context**
-

**Comment by sjmgarnier** [comment]
@mirickmi Can you post the entire installation message after you run `remotes::install_github("swarm-lab/Rvision")`? I want to see if something goes wrong before. Thanks.

**Comment by sjmgarnier** [comment]
@mirickmi Also, can you confirm that OpenCV was compiled and installed properly by running `ROpenCVLite::OpenCVPath()`? Thanks.

**Comment by mirickmi** [comment]
@sjmgarnier Thanks for the quick reply!

`ROpenCVLite::OpenCVPath()` returns **"C:\\Users\\XXX\\Documents\\R\\win-library\\4.1\\ROpenCV"**

The entire installation message: 

```
Downloading GitHub repo swarm-lab/Rvision@HEAD
v  checking for file 'C:\Users\XXX\AppData\Local\Temp\Rtmpst2r9Q\remotes368c15252567\swarm-lab-Rvision-360140f/DESCRIPTION' (582ms)
-  preparing 'Rvision': (10.9s)
v  checking DESCRIPTION meta-information ... 
-  cleaning src
-  checking for LF line-endings in source and make files and shell scripts (1.2s)
-  checking for empty or unneeded directories (368ms)
   Removed empty directory 'Rvision/man/figures'
-  building 'Rvision_0.6.2.tar.gz'
   Warnung: file 'Rvision/configure' did not have execute permissions: corrected
   
Installiere Paket nach ‘C:/Users/XXX/Documents/R/win-library/4.1’
(da ‘lib’ nicht spezifiziert)
* installing *source* package 'Rvision' ...
** using non-staged installation via StagedInstall field
+ set -e
+ C:/PROGRA~1/R/R-41~1.1/bin/x64/Rscript.exe -e 'if (nzchar(Sys.getenv('\''CI'\'')) & !ROpenCVLite::isOpenCVInstalled()) ROpenCVLite::installOpenCV(batch = TRUE)'
+ C:/PROGRA~1/R/R-41~1.1/bin/x64/Rscript.exe -e 'if (nzchar(Sys.getenv('\''CI'\'')) & ROpenCVLite::isOpenCVInstalled() & unlist(strsplit(as.character(utils::packageVersion('\''ROpenCVLite'\'')), '\''[.]'\''))[3] != gsub('\''[^0-9]'\'', '\'''\'', ROpenCVLite::opencvVersion())) ROpenCVLite::installOpenCV(batch = TRUE)'
+ C:/PROGRA~1/R/R-41~1.1/bin/x64/Rscript.exe -e 'if (!ROpenCVLite::isOpenCVInstalled()) { cat('\''------------------ OPENCV NOT FOUND --------------------\n'\'') ; cat('\''\n'\'') ; cat('\''OpenCV was not found in your library. Please install OpenCV as follows:'\'') ; cat('\''\n'\'') ; cat('\''\n'\'') ; cat('\''ROpenCVLite::installOpenCV(batch = TRUE)\n'\'') ; cat('\''\n'\'') ; cat('\''--------------------------------------------------------\n'\'') ; cat('\''\n'\'') ; stop('\''OpenCV not found.'\'') }'
** libs

*** arch - i386
"C:/rtools40/mingw32/bin/"g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-41~1.1/include" -DNDEBUG -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\include\opencv2" -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\include" -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\x86\mingw\bin" -I'C:/Users/XXX/Documents/R/win-library/4.1/Rcpp/include' -I'C:/Users/XXX/Documents/R/win-library/4.1/RcppArmadillo/include'     -I"C:/Users/XXX/Documents/R/win-library/4.1/Rcpp/include"   -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c RcppExports.cpp -o RcppExports.o
"C:/rtools40/mingw32/bin/"g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-41~1.1/include" -DNDEBUG -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\include\opencv2" -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\include" -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\x86\mingw\bin" -I'C:/Users/XXX/Documents/R/win-library/4.1/Rcpp/include' -I'C:/Users/XXX/Documents/R/win-library/4.1/RcppArmadillo/include'     -I"C:/Users/XXX/Documents/R/win-library/4.1/Rcpp/include"   -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c visionModule.cpp -o visionModule.o
In file included from visionModule.cpp:13:
Image.h: In function 'bool _writeMulti(std::__cxx11::string, Rcpp::List)':
Image.h:579:30: warning: comparison of integer expressions of different signedness: 'unsigned int' and 'R_xlen_t' {aka 'int'} [-Wsign-compare]
   for (unsigned int i = 0; i < imgList.size(); i++) {
                            ~~^~~~~~~~~~~~~~~~
In file included from visionModule.cpp:312:
shape.h: In function 'double _contourArea(Rcpp::NumericVector, Rcpp::NumericVector, bool)':
shape.h:44:22: warning: comparison of integer expressions of different signedness: 'uint' {aka 'unsigned int'} and 'R_xlen_t' {aka 'int'} [-Wsign-compare]
   for (uint i = 0; i < x.size(); i++) {
                    ~~^~~~~~~~~~
C:/rtools40/mingw32/bin/g++ -shared -s -static-libgcc -o Rvision.dll tmp.def RcppExports.o visionModule.o -LC:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\x86\mingw\lib -lopencv_calib3d460 -lopencv_core460 -lopencv_features2d460 -lopencv_flann460 -lopencv_gapi460 -lopencv_highgui460 -lopencv_imgcodecs460 -lopencv_imgproc460 -lopencv_ml460 -lopencv_objdetect460 -lopencv_photo460 -lopencv_stitching460 -lopencv_video460 -lopencv_videoio460 -LC:/PROGRA~1/R/R-41~1.1/bin/i386 -lR
installing to C:/Users/XXX/Documents/R/win-library/4.1/Rvision/libs/i386

*** arch - x64
"C:/rtools40/mingw64/bin/"g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-41~1.1/include" -DNDEBUG -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\include\opencv2" -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\include" -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\x64\mingw\bin" -I'C:/Users/XXX/Documents/R/win-library/4.1/Rcpp/include' -I'C:/Users/XXX/Documents/R/win-library/4.1/RcppArmadillo/include'     -I"C:/Users/XXX/Documents/R/win-library/4.1/Rcpp/include"   -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c RcppExports.cpp -o RcppExports.o
"C:/rtools40/mingw64/bin/"g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-41~1.1/include" -DNDEBUG -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\include\opencv2" -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\include" -I"C:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\x64\mingw\bin" -I'C:/Users/XXX/Documents/R/win-library/4.1/Rcpp/include' -I'C:/Users/XXX/Documents/R/win-library/4.1/RcppArmadillo/include'     -I"C:/Users/XXX/Documents/R/win-library/4.1/Rcpp/include"   -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c visionModule.cpp -o visionModule.o
C:/rtools40/mingw64/bin/g++ -shared -s -static-libgcc -o Rvision.dll tmp.def RcppExports.o visionModule.o -LC:\Users\XXX\DOCUME~1\R\WIN-LI~1\4.1\ROpenCV\x64\mingw\lib -lopencv_calib3d460 -lopencv_core460 -lopencv_features2d460 -lopencv_flann460 -lopencv_gapi460 -lopencv_highgui460 -lopencv_imgcodecs460 -lopencv_imgproc460 -lopencv_ml460 -lopencv_objdetect460 -lopencv_photo460 -lopencv_stitching460 -lopencv_video460 -lopencv_videoio460 -LC:/PROGRA~1/R/R-41~1.1/bin/x64 -lR
installing to C:/Users/XXX/Documents/R/win-library/4.1/Rvision/libs/x64
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
Creating a new generic function for 'sum' in package 'Rvision'
** help
*** installing help indices
  converting help for package 'Rvision'
    finding HTML links ... done
    Image-class                             html  
    LUT                                     html  
    Queue-class                             html  
    Rvision                                 html  
    RvisionAck                              html  
    Stream-class                            html  
    Video-class                             html  
    VideoWriter-class                       html  
    adaptiveThreshold                       html  
    addWeighted                             html  
    api                                     html  
    arcLength                               html  
    as.array.Rcpp_Image                     html  
    autothreshold                           html  
    bilateralFilter                         html  
    bitdepth                                html  
    blur                                    html  
    border                                  html  
    boxFilter                               html  
    canny                                   html  
    capacity                                html  
    cc_table                                html  
    changeBitDepth                          html  
    changeColorSpace                        html  
    click                                   html  
    cloneImage                              html  
    codec                                   html  
    col2bgr                                 html  
    colorspace                              html  
    compare                                 html  
    computeECC                              html  
    connectedComponents                     html  
    contourArea                             html  
    convexHull                              html  
    convexityDefects                        html  
    countNonZero                            html  
    destroyDisplay                          html  
    dim.Rcpp_Image                          html  
    dim.Rcpp_Queue                          html  
    dim.Rcpp_Stream                         html  
    dim.Rcpp_Video                          html  
    dim.Rcpp_VideoWriter                    html  
    display                                 html  
    distanceTransform                       html  
    drawArrow                               html  
    drawCircle                              html  
    drawEllipse                             html  
    drawLine                                html  
    drawPolyline                            html  
    drawRectangle                           html  
    drawRotatedRectangle                    html  
    drawText                                html  
    extractChannel                          html  
    farneback                               html  
    fillConvexPoly                          html  
    fillPoly                                html  
    filter2D                                html  
    findContours                            html  
    findNonZero                             html  
    findTransformECC                        html  
    findTransformORB                        html  
    fitEllipse                              html  
    flip                                    html  
    floodFill                               html  
    fourcc                                  html  
    fps                                     html  
    frame                                   html  
    full                                    html  
    gaussianBlur                            html  
    getPerspectiveTransform                 html  
    getTextSize                             html  
    grabCut                                 html  
    histEq                                  html  
    histmatch                               html  
    houghCircles                            html  
    huInvariants                            html  
    image                                   html  
REDIRECT:topic	 Previous alias or file overwritten by alias: C:/Users/XXX/Documents/R/win-library/4.1/Rvision/help/image.html
    imageArithmetic                         html  
    imageLogic                              html  
    imageMinMax                             html  
    imhist                                  html  
    inPlaceArithmetic                       html  
    inPlaceComparison                       html  
    inPlaceLogical                          html  
    inRange                                 html  
    inpaint                                 html  
    insertChannel                           html  
    invert                                  html  
    isBlob                                  html  
    isImage                                 html  
    isQueue                                 html  
    isStream                                html  
    isVideo                                 html  
    isVideoWriter                           html  
    laplacian                               html  
    matchShapes                             html  
    matchTemplate                           html  
    mean.Rcpp_Image                         html  
    mean.list                               html  
    medianBlur                              html  
    merge                                   html  
    min.Rcpp_Image                          html  
    minAreaRect                             html  
    minMaxLoc                               html  
    moments                                 html  
    morph                                   html  
    newDisplay                              html  
    nrow.Rcpp_Image                         html  
    ones                                    html  
    pget                                    html  
    pixelsInContour                         html  
    plot.Image                              html  
    plot.blob                               html  
    plotOF                                  html  
    pset                                    html  
    queue                                   html  
REDIRECT:topic	 Previous alias or file overwritten by alias: C:/Users/XXX/Documents/R/win-library/4.1/Rvision/help/queue.html
    queue_dimensions                        html  
    randn                                   html  
    randu                                   html  
    readFrame                               html  
    readHIS                                 html  
    readMulti                               html  
    readNext                                html  
    release                                 html  
    reset                                   html  
    resize                                  html  
    rotateScale                             html  
    scharr                                  html  
    selectROI                               html  
    sepFilter2D                             html  
    setProp                                 html  
    setTo                                   html  
    simpleBlobDetector                      html  
    sobel                                   html  
    spatialGradient                         html  
    split                                   html  
    sqrBoxFilter                            html  
    stream                                  html  
REDIRECT:topic	 Previous alias or file overwritten by alias: C:/Users/XXX/Documents/R/win-library/4.1/Rvision/help/stream.html
    stream_dimensions                       html  
    sub-.Rcpp_Image                         html  
    subImage                                html  
    sum                                     html  
    threshold                               html  
    timelapse                               html  
    video                                   html  
REDIRECT:topic	 Previous alias or file overwritten by alias: C:/Users/XXX/Documents/R/win-library/4.1/Rvision/help/video.html
    videoWriter                             html  
REDIRECT:topic	 Previous alias or file overwritten by alias: C:/Users/XXX/Documents/R/win-library/4.1/Rvision/help/videoWriter.html
    video_dimensions                        html  
    videowriter_dimensions                  html  
    warpAffine                              html  
    warpPerspective                         html  
    watershed                               html  
    write.Image                             html  
    writeFrame                              html  
    writeMulti                              html  
    writerOuput                             html  
    zeros                                   html  
** building package indices
** installing vignettes
** testing if installed package can be loaded
*** arch - i386
Error: Laden von Paket oder Namensraum für 'Rvision' in inDL(x, as.logical(local), as.logical(now), ...): fehlgeschlagen
 kann shared object 'C:/Users/XXX/Documents/R/win-library/4.1/Rvision/libs/i386/Rvision.dll' nicht laden:
  LoadLibrary failure:  Das angegebene Modul wurde nicht gefunden.

Fehler: loading failed
Ausführung angehalten
*** arch - x64
Error: Laden von Paket oder Namensraum für 'Rvision' in inDL(x, as.logical(local), as.logical(now), ...): fehlgeschlagen
 kann shared object 'C:/Users/XXX/Documents/R/win-library/4.1/Rvision/libs/x64/Rvision.dll' nicht laden:
  LoadLibrary failure:  Das angegebene Modul wurde nicht gefunden.

Fehler: loading failed
Ausführung angehalten
ERROR: loading failed for 'i386', 'x64'
* removing 'C:/Users/XXX/Documents/R/win-library/4.1/Rvision'
Warning message:
In i.p(...) :
  installation of package ‘C:/Users/XXX/AppData/Local/Temp/Rtmpst2r9Q/file368c8b356f/Rvision_0.6.2.tar.gz’ had non-zero exit status
```

**Comment by sjmgarnier** [comment]
@mirickmi Strange. The compilation part seems to go through without problem. I will need to investigate this further. That might take me a few days (I don't have access to a Windows machine at the moment). If this is urgent, you can try installing the latest version of R (4.2.1 I believe) and the corresponding version of Rtools (4.2), and then reinstall ROpenCVLite and Rvision there. That was working last time I tested it.

**Comment by mirickmi** [comment]
@sjmgarnier Thank you. No worries, I'm not in a hurry. I will install the latest version of R and report back then.

**Comment by mirickmi** [comment]
@sjmgarnier Upgrading to R version 4.2.1, then reinstalling Rtools version 4.2 and ROpenCVLite version 4.60.2 did indeed fix the problem. I could finally install Rvision 0.6.2 . Thank you!

**Comment by sjmgarnier** [comment]
I'm keeping this open to remind me to look into a fix for it.

**Comment by aharmer** [comment]
Hi, I'm having exactly the same issue as above. I will try the fix suggested but I need to use R v.4.1.3 for several other packages I may have issues. Cheers.

**Comment by sjmgarnier** [comment]
@aharmer See my reply at https://github.com/swarm-lab/ROpenCVLite/issues/28

**Comment by sjmgarnier** [comment]
This should be fixed by #60
